src/pages/Tracking/Tracking.jsx

```diff
@@ -106,6 +113,37 @@ class Tracking extends Component {
 		}
 	};
 
+	updateState = (index, value, type) => {
+		this.setState(({ deerStates }) => ({
+			deerStates: deerStates.map((deer, i) => ({
+				...deer,
+				name: i === index && type === 'name' ? value : deer.name,
+				status: i === index && type === 'status' ? value : deer.status,
+				colour: i === index && type === 'colour' ? value : deer.colour,
+				loading: {
+					status: i === index && type === 'status' ? true : deer.loading.status,
+					name: i === index && type === 'name' ? true : deer.loading.name,
+					colour: i === index && type === 'colour' ? true : deer.loading.colour,
+				},
```

Comment:
all these ternarys could probably be simplified
something like `{...deer, i===index ? ...{all the props}, : undefined}`, or maybe repalce `undefined` with `...{}` if it's an issue

Reply:
ohh nvm I see it depends on the type. there still might be an easier way maybe?